### ISSUE TITLE
Récupération du thème, producteur, mots clés dans la configuration d'une couche

### ISSUE THREAD

**Issue by elias75015** [IGNF/cartes.gouv.fr#543]
### Le besoin que vous rencontrez

Avoir les champs "keywords", "theme", "producer" facilement accessibles dans l'objet configuration (pour à terme être exploités par le service de recherche de la Géoplateforme).

Actuellement, quand via l'interface on choisit un thème pour sa couche lors de la publication d'un service, celui se retrouve dans keywords[0]

Quand on renseigne un producteur, on ne le trouve pas dans l'objet configuration.

Quand on renseigne des mots clés (libres ou non), on ne les retrouve pas dans l'objet configuration.

### La solution que vous aimeriez

Le(s) thème(s) renseigné(s) soi(en)t dans tag.theme
Le producteur renseigné soit dans tag.producer

Les mot-clés renseignés soient dans keywords

**Comment by IGNF-Xavier** [comment]
Ce n'est pas la configuration qu'il faut lire mais la métadonnée.
Il faut considérer le fait qu'il y a :
- plusieurs géoservices configurés
- plusieurs moyens d'alimenter Géoplateforme, la solution proposée complexifiera l'utilisation de l'API et du SDK.

**Comment by elias75015** [comment]
> plusieurs géoservices configurés (précision : liés à une seule métadonnée)

Justement, je parle du processus au moment de la publication d'un service. Le service de recherche (et l'entrée carto) fonctionnent en termes de services (couche WMS/WMTS/TMS...). La fiche de métadonnée en XML en plus d'être liée à une donnée n'est vraiment pas adaptée pour être parsée dans un contexte web.

> plusieurs moyens d'alimenter Géoplateforme, la solution proposée complexifiera l'utilisation de l'API et du SDK en obligeant le producteur à dupliquer des informations entre configuration et métadonnées.

Le ticket n'impose pas à l'utilisateur (cartes.gouv, SDK ou API) de remplir un producteur / theme / mots clés. La demande est juste de corriger le mapping de certaines informations pour qu'on puisse correctement les exploiter. Les inputs pour remplir ces champs sont d'ores et déjà présents dans l'interface de publication de cartes.gouv. **S'ils sont remplis** par l'utilisateur, je propose qu'on les retrouve juste dans les "tags" qui sont justement prévus pour ajouter de l'information côté entrepôt afin d'être exploités par le service de recherche de la Géoplateforme (qui pourra filtrer dessus). C'est d'ailleurs documenté ici : 
https://geoplateforme.github.io/tutoriels/production/recherche/standard/#champs-theme-licence-et-thumbnail

De plus, les keywords ne se retrouvent pas dans config.keywords : or c'est ce champ qui permet de bien renseigner le getCap avec les mots clés.

**Comment by IGNF-Xavier** [comment]
Les tags, posent vraiment problèmes dans l'usage demandé.
J'ai l'impression que nous n'avions pas complètement ciblé le sujet.
**Réunion éventuellement à prévoir.** 

Les informations de type keyword sont renseignées au travers de 2 étapes, peu importe le vecteur : 
- la première au travers des métadonnées poussées sur Geonetwork de la Géoplateforme
- la seconde au travers de la configuration puis publication du géoservice et on les retrouve dans le getcapabilities , par exemple :  https://data.geopf.fr/wfs/ows?SERVICE=WFS&VERSION=2.0.0&REQUEST=GetCapabilities
![image](https://github.com/user-attachments/assets/fc68840d-acf1-4a75-8937-b38c923fd9af)

Je ne sais pas dans quelle mesure ces keywords provenant de ces sources (métadonnées et getacapabilities) - et pas associées au concept de tags de l'API Entrepôt - sont exploitées pour peupler  le service de recherche.
Mais sauf erreur, cette exploitation est indépendante des workflow mis en place depuis cartes.gouv

Je pense donc que le sujet est plutôt au niveau de @Dolite au lieu de @slafayIGN 

Et je suis tout à fait en phase avec le besoin car il est envisagé à terme de mettre en place un outil de construction de getcapabilities en fonction des keywords renseignés dans les getcapaibilities des services.

**Comment by Dolite** [comment]
Les Keywords dans la configuration sont à destination en premier lieu du service de diffusion, pour qu'ils apparaissent dans le getcapabilities. Ils sont aussi présents dans le document dans le service de recherche.

Les tags sont un mécanisme générique de l'entrepôt, d'abord utilisé pour catégoriser les données, pouvoir les retrouver facilement par filtrage. Avec l'évolution des besoins, certains tags sur certaines entités sont exploités pour enrichir le document du service de recherche (theme, edition_year...). 

Je ne comprends pas ce qui serait de côté entrepôt : il y a des fonctionnements en place, qui ici semblent suffire pour répondre au besoin exprimé par Elias, mais ne sont juste pas exploités par cartes.gouv (les keywords se retrouvent avec la valeur du thème alors qu'un tag theme est fait pour au niveau de la stored data pour se finir au bon endroit dans le service de recherche). A priori les valeurs sont bien renseignées par la personne qui alimente en données, mais elles ne sont pas forcément dispatchées sur les bons champs dans les entités entrepôt.

**Comment by elias75015** [comment]
Le service de recherche peut taper filtrer par :
 \"shape\", \"aggregation\", \"production_year\", \"theme\", \"open\", \"is_search_layer\", \"licence\", \"highlight\", \"title\", \"layer_name\", \"type\", \"description\", \"publication_date\", \"keywords\", \"metadata_urls\", \"attribution_title\", \"attribution_email\", \"srs\"

cf 
https://geoservices.ign.fr/documentation/services/services-geoplateforme/service-geoplateforme-de-recherche#72759
et 
https://data.geopf.fr/recherche/swagger-ui/index.html#/Recherche/getCapabilities

**Comment by slafayIGN** [comment]
La topicCategory, en français, doit être poussée comme tag "theme" de la configuration c'est bien ça ?

**Comment by elias75015** [comment]
Oui, ce qui correspond à cette liste :https://github.com/IGNF/cartes.gouv.fr/blob/main/assets/data/topic_categories.json

**Comment by elias75015** [comment]
A terme on pourra faire ce filtre sur les données ayant une metadonnées associée.

**Comment by slafayIGN** [comment]
Détaillé pour keywords et metadata_urls dans #710

**Comment by IGNF-Xavier** [comment]
Attention côté Entrepôt : IGNGPF-4591
